fixups/RegLegacyFixups/RegistryFixups.cpp

```diff
@@ -257,6 +261,28 @@ REGSAM RegFixupSam(std::string keypath, REGSAM samDesired, DWORD RegLocalInstanc
     return samModified;
 }
 
+template <typename CharT>
+const char* ReplaceRegistryQueryPath(PHKEY key, const CharT* subKey)
+{
+    std::string keypath = InterpretKeyPath(*key) + "\\" + InterpretStringA(subKey);
+
+    std::string normalizedKeypath = ReplaceRegistrySyntax(keypath);
+    size_t subkeyOffset = normalizedKeypath.find_first_of('\\');
+    if (subkeyOffset != std::wstring_view::npos)
+    {
+        std::string requestedSubkey = normalizedKeypath.substr(subkeyOffset + 1);
+        auto it = g_regRedirectedHivePaths.find(requestedSubkey);
+        if (it != g_regRedirectedHivePaths.end())
+        {
+            *key = HKEY_CURRENT_USER;
+            return it->c_str();
```

Comment:
I'm iffy on returning a pointer.  If the original string is deleted the returned pointer will be invalid.

Reply:
Its a pointer to a global variable. Original string is kept throughout the lifetime of the fixup.